### ISSUE TITLE
MailSender does not launch

### ISSUE THREAD

**Issue by jefflongo** [ACRA/acra#1146]
**Describe the bug**
MailSender does not trigger after tapping notification. Notification is dismissed and exception is thrown in logcat. 

ACRA configuration:
```java
ACRA.init(this, new CoreConfigurationBuilder()
                .withBuildConfigClass(BuildConfig.class)
                .withReportFormat(StringFormat.JSON)
                .withLogcatArguments("-t", "100", "-v", "long", "-s", MYTAG1 + ":V", MYTAG2 + ":V",
                        MYTAG3 + ":V")
                .withPluginConfigurations(
                        new NotificationConfigurationBuilder()
                                .withTitle(getString(R.string.acra_notif_title))
                                .withText(getString(R.string.acra_notif_text))
                                .withChannelName(getString(R.string.app_name))
                                .withSendOnClick(true)
                                .build(),
                        new MailSenderConfigurationBuilder()
                                .withMailTo(getString(R.string.acra_developer_email))
                                .withReportAsFile(true)
                                .withReportFileName(getString(R.string.acra_file_name))
                                .withSubject(getString(R.string.acra_mail_subject))
                                .withBody(getString(R.string.acra_mail_body))
                                .build()
                )
        );
```

Logcat dump of exception:
```
Attempt to cast generated internal exception:
java.lang.ClassCastException: Cannot cast java.lang.String to java.util.ArrayList
at java.lang.Class.cast(Class.java:2479)
at android.os.BaseBundle.getValueAt(BaseBundle.java:405)
at android.os.BaseBundle.getValue(BaseBundle.java:374)
at android.os.BaseBundle.getArrayList(BaseBundle.java:1486)
at android.os.BaseBundle.getCharSequenceArrayList(BaseBundle.java:1529)
at android.os.Bundle.getCharSequenceArrayList(Bundle.java:1204)
at android.content.Intent.getCharSequenceArrayListExtra(Intent.java:9709)
at android.content.Intent.migrateExtraStreamToClipData(Intent.java:12784)
at android.app.Instrumentation.execStartActivity(Instrumentation.java:1799)
at android.app.ContextImpl.startActivity(ContextImpl.java:1145)
at android.app.ContextImpl.startActivity(ContextImpl.java:1116)
at android.content.ContextWrapper.startActivity(ContextWrapper.java:441)
at android.content.ContextWrapper.startActivity(ContextWrapper.java:441)
at org.acra.sender.EmailIntentSender.sendLegacy(EmailIntentSender.kt:104)
at org.acra.sender.EmailIntentSender.sendWithSelector(EmailIntentSender.kt:131)
at org.acra.sender.EmailIntentSender.send(EmailIntentSender.kt:64)
at org.acra.sender.ReportSender.send(ReportSender.kt:59)
at org.acra.sender.ReportDistributor.sendCrashReport(ReportDistributor.kt:101)
at org.acra.sender.ReportDistributor.distribute(ReportDistributor.kt:65)
at org.acra.sender.SendingConductor.sendReports(SendingConductor.kt:50)
at org.acra.scheduler.DefaultSenderScheduler.scheduleReportSending(DefaultSenderScheduler.kt:61)
at org.acra.scheduler.SchedulerStarter.scheduleReports(SchedulerStarter.kt:49)
at org.acra.receiver.NotificationBroadcastReceiver.onReceive(NotificationBroadcastReceiver.kt:72)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:4881)
at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2417)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:226)
at android.os.Looper.loop(Looper.java:313)
at android.app.ActivityThread.main(ActivityThread.java:8741)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

**Expected behavior**
Gmail should open and draft an email bug report.

**Version**
 - Samsung S22+
 - Android: 13 (API 33)
 - ACRA 5.9.7
 - Compile SDK version: 33
 - Target SDK version: 33
 - Java

**Comment by rethinavelsmitch** [comment]
Describe the bug:
Mail not received after displaying toast message., Working in Pixel4a and not working in OnePlus and Redmi Devices.

ACRA configuration:

  ```
  initAcra {
            // core configuration:
            buildConfigClass = BuildConfig::class.java
            reportFormat = StringFormat.JSON

            toast {
                // required
                text = "Application Crashed, Logs will be sent to Developers"
                // defaults to Toast.LENGTH_LONG
                length = Toast.LENGTH_LONG
            }

            mailSender {
                // required
                mailTo = "veluXXXXXXXXX@XXXXXXX.com"
                reportAsFile = true
                reportFileName = "Crash.txt"
                subject = BuildConfig.APPLICATION_ID + " - " + BuildConfig.VERSION_NAME
                body = "Hey Velu, I found a crash in Recent Android APK, PFA. "
            }
        }

```

Device Specifications : One Plus Nord

**Comment by henrichg** [comment]
@jefflongo
If user has installed also another e-mail application, for example Samsung E-mail, the internal ReportSender working, is possible to send report with GMail.
And is not needed to configure Samsung E-mail for this (add account).

**Comment by JeanPaulLucien** [comment]
Do Acra not work with GMail app? 
Just I did not see Acra's activity on Android 4.2.2. Acra shows popup to ask what was wrong but does not open mail app.

**Comment by yuriykulikov** [comment]
Possibly related to https://github.com/ACRA/acra/issues/1118

**Comment by JeanPaulLucien** [comment]
Works for me in Android 4.2.2. But I don't know version of ACRA.

**Comment by yuriykulikov** [comment]
I am experiencing the same issue with `5.9.7`, Pixel 6a, Android 13, Gmail. I can only see these suspicious lines in the logs:
```
2022-12-30 20:02:31.549 29618-29618 ACRA                    com.better.alarm.debug               I  Sending report /data/user/0/com.better.alarm.debug/app_ACRA-approved/2022-12-30T20:02:31.537+01:00-IS_SILENT.stacktrace
2022-12-30 20:02:31.560  1685-7546  PackageManager          pid-1685                             W  Intent does not match component's intent filter: Intent { act=android.intent.action.SEND flg=0x10000000 (has extras) sel=act=android.intent.action.SENDTO dat=mailto: flg=0x10000000} }
2022-12-30 20:02:31.560  1685-7546  PackageManager          pid-1685                             W  Access blocked: ComponentInfo{com.google.android.gm/com.google.android.gm.ComposeActivityGmailExternal}
```

**Comment by yuriykulikov** [comment]
I am experiencing the same issue with `5.9.7`, Pixel 6a, Android 13, Gmail. I can only see these suspicious lines in the logs:
```
2022-12-30 20:02:31.549 29618-29618 ACRA                    com.better.alarm.debug               I  Sending report /data/user/0/com.better.alarm.debug/app_ACRA-approved/2022-12-30T20:02:31.537+01:00-IS_SILENT.stacktrace
2022-12-30 20:02:31.560  1685-7546  PackageManager          pid-1685                             W  Intent does not match component's intent filter: Intent { act=android.intent.action.SEND flg=0x10000000 (has extras) sel=act=android.intent.action.SENDTO dat=mailto: flg=0x10000000} }
2022-12-30 20:02:31.560  1685-7546  PackageManager          pid-1685                             W  Access blocked: ComponentInfo{com.google.android.gm/com.google.android.gm.ComposeActivityGmailExternal}
```

**Comment by F43nd1r** [comment]
@NLLAPPS that looks more like your app crashed outside of the context of an activity, in which case it won't be allowed to start anything in the foreground.

**Comment by yuriykulikov** [comment]
This PR https://github.com/ACRA/acra/pull/1170 by @grueni75 solves the issues on my devices (GMail is the only client). Thank you, @grueni75!

**Comment by yuriykulikov** [comment]
@F43nd1r what was the reason for using `intent.selector`? Was it because of gmail in 2021?

Perhaps the `legacy` function can replace the selector, with a minor tweak to get rid of deprecated use:

```
  private fun queryActivities(context: Context): MutableList<ResolveInfo> {
    val pm = context.packageManager
    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
      pm.queryIntentActivities(
          buildSendToIntent(),
          PackageManager.ResolveInfoFlags.of(PackageManager.MATCH_DEFAULT_ONLY.toLong()))
    } else {
      @Suppress("DEPRECATION")
      pm.queryIntentActivities(buildSendToIntent(), PackageManager.MATCH_DEFAULT_ONLY)
    }
  }
```

**Comment by F43nd1r** [comment]
Everybody please test with 5.9.8-beta01 - I have removed using selector intents on tiramisu. Need data on how this changes things for you.

**Comment by jefflongo** [comment]
> 

Just tried 5.9.8-beta02, still not working for me. Same error as in the OP with just gmail installed.

**Comment by F43nd1r** [comment]
@jefflongo  the stacktrace in OPs post is a red herring. This error is produced by android but it continues on to handle the string correctly as a string. If it's not working for you, look for other errors or warnings.

**Comment by jefflongo** [comment]
The only thing that comes after that is this:
```
Background activity start [callingPackage: com.jefflongo.redacted; callingUid: 10490; appSwitchState: 2; isCallingUidForeground: false; callingUidHasAnyVisibleWindow: false; callingUidProcState: RECEIVER; isCallingUidPersistentSystemProcess: false; realCallingUid: 10490; isRealCallingUidForeground: false; realCallingUidHasAnyVisibleWindow: false; realCallingUidProcState: RECEIVER; isRealCallingUidPersistentSystemProcess: false; originatingPendingIntent: null; allowBackgroundActivityStart: false; intent: Intent { act=android.intent.action.SEND_MULTIPLE typ=*/* flg=0x10000001 pkg=com.google.android.gm cmp=com.google.android.gm/.ComposeActivityGmailExternal clip={*/* {U(content)}} (has extras) }; callerApp: ProcessRecord{d0dded8 30892:com.jefflongo.longbow:acra/u0a490}; inVisibleTask: false]
```
and gmail does not launch. Sometimes when I re-open the app after the crash the notification will appear again and clicking it will sometimes open gmail.

**Comment by jefflongo** [comment]
Ok @F43nd1r I found some more information. There is a `Indirect notification activity start (trampoline) from com.example.acracrashdemo blocked` error in logcat after tapping "ok" on the notification to launch gmail. I was able to reproduce this with this [demo application](https://github.com/jefflongo/acra-crash-demo) I made on an API 32 resizable emulator device. It's worth noting that on the API 33 resizable emulator device it worked properly.
